### PR TITLE
Fix #972: Migrate TPP client to Rest Client

### LINKED ISSUE #972: Migrate TPP client to Rest Client
The TPP engine should be migrated to Rest client from RestTemplate.

### PR DESCRIPTION
**Pull request by romanstrobl**

### CODE REVIEW

File: powerauth-tpp-engine-client/src/main/java/io/getlime/security/powerauth/lib/tpp/engine/client/TppEngineClient.java [hunk 1]
```diff
@@ -31,89 +34,62 @@
...
+     * @param serviceUrl Base service URL.
+     * @throws TppEngineClientException Thrown when REST client initialization fails.
      */
-    public TppEngineClient(String serviceUrl, ObjectMapper objectMapper) {
-        this.serviceUrl = serviceUrl;
-        if (objectMapper != null) {
-            this.objectMapper = objectMapper;
-        } else {
-            this.objectMapper = objectMapper();
+    public TppEngineClient(String serviceUrl) throws TppEngineClientException {
```
Comment (petrdvorak) [inline powerauth-tpp-engine-client/src/main/java/io/getlime/security/powerauth/lib/tpp/engine/client/TppEngineClient.java:65]: I think we need to keep an option to pass `ObjectMapper` here - maybe just introduce a wrapper method?

**Review by petrdvorak — APPROVED** [review]
Looks OK, I would just add constructor that allows setting a custom ObjectMapper...

File: powerauth-tpp-engine-client/src/main/java/io/getlime/security/powerauth/lib/tpp/engine/client/TppEngineClient.java (hunk 1 above)
Comment (romanstrobl) [reply powerauth-tpp-engine-client/src/main/java/io/getlime/security/powerauth/lib/tpp/engine/client/TppEngineClient.java:65]: We can use the constructor `TppEngineClient(RestClientConfiguration config)`, which allows setting the custom `ObjectMapper` as a parameter. I think this solves are customization use cases.

Comment (petrdvorak) [reply powerauth-tpp-engine-client/src/main/java/io/getlime/security/powerauth/lib/tpp/engine/client/TppEngineClient.java:65]: 👍  Yes, this works for me.

**Pull request merged**